code/workspaces/web-app/src/components/modal/SiteLogs.js

```diff
@@ -1,28 +1,56 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import Dialog from '@material-ui/core/Dialog';
 import DialogTitle from '@material-ui/core/DialogTitle';
 import DialogContent from '@material-ui/core/DialogContent';
 import DialogActions from '@material-ui/core/DialogActions';
 import DialogContentText from '@material-ui/core/DialogContentText';
 import PrimaryActionButton from '../common/buttons/PrimaryActionButton';
+import PromisedContentWrapper from '../common/PromisedContentWrapper';
 
-const Logs = ({ title, body, onCancel }) => (
-  <Dialog open={true} onClose={onCancel} maxWidth="md">
-    <DialogTitle>{title}</DialogTitle>
-    <DialogContent>
-      <DialogContentText style={{ whiteSpace: 'pre' }}>{body}</DialogContentText>
-    </DialogContent>
-    <DialogActions>
-      <PrimaryActionButton onClick={onCancel}>Close</PrimaryActionButton>
-    </DialogActions>
-  </Dialog>
-);
+class Logs extends Component {
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      logs: { fetching: true },
+    };
+  }
+
+  async componentDidMount() {
+    let logs = await this.props.getLogs(this.props.projectName, this.props.stackName);
+    if (logs.value === '') {
+      logs = { value: 'ERROR: Unable to retrieve logs.' };
+    }
```

Comment:
Empty strings are `falsey` in JS so you could just have `if (!logs.value)` which would also catch when this is `null`, `undefined`, etc.

Is it true to say that there has been an error if the value is an empty string? What happens if there just aren't any logs?

Reply:
Ah right thanks for that. Well all of the stacks do have startup logs so I think it's an okay assumption for the moment.

I'm not sure what another way of handling all errors would be, the response from the API for a non-existent stack is technically something like this

```
...
"errors": [
    {
      "message": "Request failed with status code 404",
...
"data": {
    "logs": null
  }
...
```